FirebaseMessaging/Sources/FIRMessagingAnalytics.m

```diff
@@ -212,11 +216,34 @@ + (void)logMessage:(NSDictionary *)notification
       break;
 
     default:
-      // Only a silent notification (i.e. 'content-available' is true) can be received while the app
-      // is in the background. These messages aren't loggable anyway.
+      // App was either in background state or in transition from closed
+      // to open.
+      // Needs to check notification is displayed.
+      if ([[self class] isDisplayNotification:notification]) {
+        [self logOpenNotification:notification toAnalytics:analytics];
+      }
       break;
   }
 #endif
 }
 
++ (BOOL)isDisplayNotification:(NSDictionary *)notification {
```

Comment:
Would you mind adding a comment to describe the expected dictionary format e.g. a link to docs or with a format like `@{ kApsKey : @{ kApsAlertKey : @"alert" } }`. I think it will be useful for future us.